server/src/main/java/com/linecorp/centraldogma/server/internal/httpapi/ContentServiceV1.java

```diff
@@ -0,0 +1,345 @@
+import com.linecorp.centraldogma.common.EntryType;
+import com.linecorp.centraldogma.common.Markup;
+import com.linecorp.centraldogma.common.Query;
+import com.linecorp.centraldogma.common.Revision;
+import com.linecorp.centraldogma.internal.Jackson;
+import com.linecorp.centraldogma.internal.httpapi.v1.CommitMessageDto;
+import com.linecorp.centraldogma.internal.httpapi.v1.EntryDto;
+import com.linecorp.centraldogma.server.internal.admin.authentication.AuthenticationUtil;
+import com.linecorp.centraldogma.server.internal.command.Command;
+import com.linecorp.centraldogma.server.internal.command.CommandExecutor;
+import com.linecorp.centraldogma.server.internal.storage.project.ProjectManager;
+import com.linecorp.centraldogma.server.internal.storage.repository.ChangeConflictException;
+import com.linecorp.centraldogma.server.internal.storage.repository.FindOption;
+import com.linecorp.centraldogma.server.internal.storage.repository.Repository;
+
+public class ContentServiceV1 extends AbstractService {
+
+    public ContentServiceV1(ProjectManager projectManager, CommandExecutor executor) {
+        super(projectManager, executor);
+    }
+
+    /**
+     * GET /projects/{projectName}/repos/{repoName}/tree{path}?revision={revision}
+     *
+     * <p>Returns the list of files in the path.
+     */
+    @Get("regex:/projects/(?<projectName>[^/]+)/repos/(?<repoName>[^/]+)/tree(?<path>/.*$)")
+    public CompletionStage<List<EntryDto>> listFiles(@Param("projectName") String projectName,
+                                                     @Param("repoName") String repoName,
+                                                     @Param("path") String path,
+                                                     @Param("revision") @Default("-1") String revision) {
+        checkRepositoryExists(projectName, repoName);
+
+        final Repository repository = getRepository(projectName, repoName);
+        return listFiles(repository, path, new Revision(revision),
+                         ImmutableMap.of(FindOption.FETCH_CONTENT, false));
+    }
+
+    private CompletionStage<List<EntryDto>> listFiles(Repository repository, String path,
+                                                      Revision revision, Map<FindOption<?>, ?> options) {
+        final String pathPattern;
+        if (path.endsWith("/")) {
+            pathPattern = path + '*';
+        } else if (!path.endsWith("*")) {
+            pathPattern = path + "/*";
+        } else {
+            pathPattern = path;
+        }
+
+        return repository.find(revision, pathPattern, options)
+                         .thenApply(entries -> entries.values().stream()
+                                                      .filter(entry -> entry.type() != EntryType.DIRECTORY)
+                                                      .map(DtoConverter::convert).collect(toList()));
+    }
+
+    /**
+     * POST /projects/{projectName}/repos/{repoName}/contents{path}?revision={revision}
```

Comment:
Looks different from the implementation. Which one is correct?

Reply:
I didn't change the path when it's changed from PUT to POST, thank you~!